src/Highlighter.ts

```diff
@@ -113,18 +119,34 @@ export default class Highlighter {
     return this.container.ownerDocument;
   }
 
-  private onMouseup = (ev: MouseEvent): void => {
+  private onSelectionChange = (): void => {
     const selection = this.document.getSelection();
 
-    if (!selection) {
+    if (
+      !selection
+      || selection.isCollapsed
+      || selection.type === 'None'
+      || !dom(this.container).contains(selection.anchorNode)
+      || !dom(this.container).contains(selection.focusNode)
+      || this.compareRanges(selection ? getRange(selection) : null, this.previousRange)
```

Comment:
whats with the whole previousRange thing?

Reply:
In `onSelect` I'm setting `this.previousRange` to snapped range
since snapping is triggering `selectionchange` event I want to noop in this case